src/main/java/org/opensearch/performanceanalyzer/decisionmaker/actions/SearchBackPressureAction.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.decisionmaker.actions;
+
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.annotations.SerializedName;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.performanceanalyzer.AppContext;
+import org.opensearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+
+public class SearchBackPressureAction extends SuppressibleAction {
+    private static final Logger LOG = LogManager.getLogger(SearchBackPressureAction.class);
+    public static final String NAME = "SearchBackPressureAction";
+    private static final ImpactVector NO_IMPACT = new ImpactVector();
+
+    /*
+     * Time to wait since last recommendation, before suggesting this action again
+     */
+    private static final long DEFAULT_COOL_OFF_PERIOD_IN_MILLIS = TimeUnit.DAYS.toMillis(1);
+
+    /* From Config Per Diumension Type
+     *  canUpdate: whether the action should be emitted
+     *  coolOffPeriodInMillis: how long the CoolOffPeriod the action should before reemit
+     *  thresholdName: the name of threshold we are tuning  (e.g. node_duress.cpu_threshold, search_heap_threshold)
+     *  dimension: indicates whether the resource unit is caused by shard/task level searchbackpressure cancellation stats
+     *  Step Size in percentage: how much should the threshold change in percentage
+     */
+    private boolean canUpdate;
+    private long coolOffPeriodInMillis;
+    private String thresholdName;
+
+    // TODO: change dimension, direction as enum
+    private SearchbpDimension dimension;
+    private SearchbpThresholdActionDirection direction;
```

Comment:
Is the SB tuning bi-directional ?

Reply:
Yes, can either Increase or Decrease

Can be inferred from the SearchbpThresholdActionDirection enum

 ```
public enum SearchbpThresholdActionDirection {
        INCREASE(SearchbpThresholdActionDirection.Constants.INCREASE),
        DECREASE(SearchbpThresholdActionDirection.Constants.DECREASE);

        private final String value;

        SearchbpThresholdActionDirection(String value) {
            this.value = value;
        }

        @Override
        public String toString() {
            return value;
        }

        public static class Constants {
            public static final String INCREASE = "increase";
            public static final String DECREASE = "decrease";
        }
    }
```